tactics/declare.mli

```diff
@@ -32,12 +103,26 @@ type 'a proof_entry = private {
   proof_entry_inline_code : bool;
 }
 
+(** XXX: Scheduled for removal from public API, don't rely on it *)
```

Comment:
In particular making this private will allow us address longstanding issues with the design of `start_dependent_proof` API and side-effects.

Reply:
>Scheduled for removal from public API

This is the definition of deprecated, so why don't we deprecate it?

Reply:
Usually we have followed the rule that when we deprecate stuff we don't allow uses of it in the codebase, except for a few exceptions; I can do that, but actually part two makes this go away and there is no chance of compat as we'd like to modify the internals.

So I've opted for the path to remove it directly without adding a deprecation phase that actually doesn't remove the uses in the codebase. I can go any other route but IMVHO the most efficient path is to try to consolidate all the declaration modules in one ASAP.

I already ported the remaining uses in the codebase so only thing that remains to solve is the fact that `declareDef/Obl ` are in vernac and `declare` is in tactics; this requires some cleanup on schemes so it will take a bit of extra time but not a lot [should make into 8.12]